### ISSUE TITLE
Custom scrollbars affecting functionality 

### ISSUE THREAD

**Issue by narcolepticinsomniac** [mauricecruz/chrome-devtools-zerodarkmatrix-theme#152]
Hey Maurice. The limitation you explained on how the CSS can be injected is never gonna allow for custom scrollbars to render with any degree of efficiency. More often they don't, or one does and the other doesn't. None of this would be a big deal, but whatever code you're using is causing the default scrollbars to malfunction. Part of the width is cut off and what's  worse is the buttons and the area directly under/over them don't work. Once it starts malfunctioning, you can't even grab the thumb and drag it correctly. 

I'm a huge fan of this style/extension, but it's getting to the point where this bug is making it too annoying to use. Unfortunately, short of a better injection method, I think scrapping them altogether or (maybe) having all the specs from yours match the default are the only options.

**Comment by mauricecruz** [comment]
Thanks for the detailed description of the issue. I'd like to do some investigation. Can you provide the chrome version and OS you're using?

If possible, also a screenshot to reference when testing.

**Comment by narcolepticinsomniac** [comment]
Latest stable: Version 47.0.2526.80 m (64-bit). It's not easily reproducible. It seems to only happen when you're in the middle of something and it's a huge pain in the ass. 

The part about them not rendering correctly is easy to reproduce, it's a crap-shoot when Devtools first open. The buggish behavior happens occasionally and the only explanation is that the defaults are showing but some of your CSS is being applied to functionality. Most notably:

 ::-webkit-scrollbar-button {
display:none!important;
}

I'm guessing.

**Comment by nikolowry** [comment]
Was about to add a new suggestion as an issue, but since it deals with scrollbar functionality I'll add it here:

Any interest in detecting overlay scrollbars and then consequently not theming the scrollbars if detected? 

Using the tried and trusted method that David Walsh wrote about a few years back would do the trick [https://davidwalsh.name/detect-scrollbar-width].  So any scrollbars that return a scrollbar width === 0 wouldn't need any styling.  I'm assuming this would mean having the scrollbar styles in it's own CSS file so it can be applied independently of the theme.

**Comment by mauricecruz** [comment]
Sorry for the delay but I was finally able to replicate the issue! The theming api doesn't let me use any js to do any checks or calculations so it might be a bit tricky. But I'll figure out a solution.

**Comment by nikolowry** [comment]
@mauricecruz there might be couple options, but they require you managing the scrollbars css and the theme css independently:

- check the platform (Chromium v50 on OSX doesn't have a chrome-flag for "overlay scrollbars" anymore, so I assume Linux/Windows are the only ones that might have it now and OSX uses overlay by default) and if the platform is Linux/Windows, check for to see if the flag (is there an easy way of doing that yet?) is enabled.  Then you either combine the previously proposed scrollbars.css/theme.css or omit the scrollbars.css depending on the user preference.

- If there isn't a reliable way to check for flags, there is one other way to pull it off but it requires you adding a devtools panel and then creating a "bridge" between your panel and the devtools.js.  Then you can use the david walsh test and update the theme stylesheet between injecting it. It be kind of similar to how the "devtools author themes extension" is architected.

**Comment by narcolepticinsomniac** [comment]
Glad to hear it. The bug probably pops up more depending on how you use devtools. It was happening quite often for me, so I finally had to disable your theme. 

I can see why you don't want to leave the default light scrollbars in a dark theme. If your main objective is to make them dark, I came up with a hackish solution. I made my own stylesheet and have been sideloading it. It's nowhere near as complete as yours (especially canary), but maybe give it a whirl. A glimpse of the CSS will show that the method is not ideal, but don't knock it till you try it. It works pretty well.

[devtools-css.txt](https://github.com/mauricecruz/chrome-devtools-zerodarkmatrix-theme/files/143634/devtools-css.txt)


I like your color scheme well enough and you do a nice job keeping it updated, plus I hate the sideload warning, so I'd much rather switch back to yours once it's sorted out.
